src/chainlit/socket.py

```diff
@@ -85,6 +100,9 @@ def ask_user_fn(data, timeout):
 @socket.on("connection_successful")
 async def connection_successful(sid):
     session = Session.require(sid)
+    if session.restored:
```

Comment:
we could merge but I initially did this for the following reason:

In `connection_successful`  we potentially call `on_chat_start` which might send messages to the UI. If we do that in the `connect` handler, the connection is not yet accepted in my understanding (since it will be accepted if we return True).

Sending messages while the connection is not accepted seems weird to me.